TWTValidation/JSON Validator/Processors/TWTJSONObjectValidatorGenerator.m

```diff
@@ -0,0 +1,410 @@
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions:
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+//
+
+#import <TWTValidation/TWTJSONObjectValidatorGenerator.h>
+
+#import <TWTValidation/TWTValidation.h>
+
+#import <TWTValidation/TWTJSONSchemaASTCommon.h>
+#import <TWTValidation/TWTJSONSchemaParser.h>
+#import <TWTValidation/TWTJSONSchemaArrayValidator.h>
+#import <TWTValidation/TWTJSONSchemaObjectValidator.h>
+
+
+@interface TWTJSONObjectValidatorGenerator ()
+
+@property (nonatomic, strong, readonly) NSMutableArray *objectStack;
+
+@end
+
+
+@implementation TWTJSONObjectValidatorGenerator
+
+- (instancetype)init
+{
+    self = [super init];
+    if (self) {
+        _objectStack = [[NSMutableArray alloc] init];
+    }
+    return self;
+}
+
+
+- (TWTJSONObjectValidator *)validatorFromJSONSchema:(NSDictionary *)schema error:(NSError *__autoreleasing *)outError warnings:(NSArray *__autoreleasing *)outWarnings
+{
+    [self.objectStack removeAllObjects];
+
+    TWTJSONSchemaParser *parser = [[TWTJSONSchemaParser alloc] initWithJSONSchema:schema];
+    NSError *parsingError = nil;
+    TWTJSONSchemaTopLevelASTNode *topLevelNode = [parser parseWithError:&parsingError warnings:outWarnings];
+    if (parsingError) {
+        if (outError) {
+            *outError = parsingError;
+        }
+        return nil;
+    }
+
+    [topLevelNode acceptProcessor:self];
+    return [self popCurrentObject];
+}
+
+
+#pragma mark - ASTNodeProcessor protocol methods
+
+- (void)processTopLevelNode:(TWTJSONSchemaTopLevelASTNode *)topLevelNode
+{
+    [topLevelNode.schema acceptProcessor:self];
+}
+
+
+- (void)processGenericNode:(TWTJSONSchemaGenericASTNode *)genericNode
+{
+    TWTValidator *commonValidator = [self commonValidatorFromNode:genericNode];
+    TWTValidator *typeValidator = nil;
+
+    if ([genericNode.validTypes containsObject:TWTJSONSchemaTypeKeywordBoolean]) {
+        typeValidator = [[TWTBlockValidator alloc] initWithBlock:^BOOL(id value, NSError *__autoreleasing *outError) {
+            return [value isKindOfClass:[NSNumber class]] && strcmp([(NSValue *)value objCType], @encode(BOOL)) == 0;
+        }];
```

Comment:
Actually, you don’t need to, but leave a comment saying what this does. Also, cast value to an NSNumber, not an NSValue.